src/runtime_strings.js

```diff
@@ -148,7 +148,14 @@ function UTF8ArrayToString(u8Array, idx, maxBytesToRead) {
  * @return {string}
  */
 function UTF8ToString(ptr, maxBytesToRead) {
+#if TEXTDECODER == 2
+  if (!ptr) return '';
+  var maxPtr = ptr + maxBytesToRead;
+  for(var end = ptr; !(end >= maxPtr) && HEAPU8[end];) ++end;
+  return UTF8Decoder.decode(HEAPU8.subarray(ptr, end));
```

Comment:
I don't know if we still have value from the `heapAndOffset` helper function used 2 lines down. But we should at least use the same method in both. Just doing `HEAP8.subarray` in both sounds ok.

Reply:
Check, that makes sense.